stmhal/timer.c

```diff
@@ -1095,9 +1104,12 @@ STATIC mp_obj_t pyb_timer_callback(mp_obj_t self_in, mp_obj_t callback) {
         __HAL_TIM_DISABLE_IT(&self->tim, TIM_IT_UPDATE);
         self->callback = mp_const_none;
     } else if (mp_obj_is_callable(callback)) {
+        __HAL_TIM_DISABLE_IT(&self->tim, TIM_IT_UPDATE);
```

Comment:
This is just disabling the update timer (not all interrupts). Otherwise, I found that just the act of calling HAL_TIM_BASE_Start_IT could cause the interrupt handler to immediately fire.

I guess I don't see the issue with missing an interrupt. Under this scenario:

Initialize the timer
wait some time so that the interrupt would have fired but didn't because there was no callback
Set the callback

then yes you would miss that interrupt. But you could have avoided that by setting the callback when the timer was initialized rather than setting it afterwards.

With most of the timer interrupts, having an extraneous interrupt at the beginning doesn't really matter. With the encoder, it means that your counter overflowed/underflowed, and I was getting an interrupt being fired 100% of the time here and then again when the HAL_TIM_Encoder_Init function was called (so if I passed a callback into the timer init it would fire twice before it even started counting).